docs/docs/prerender.md

```diff
@@ -49,23 +49,55 @@ You can also prerender your not found page (a.k.a your 404 page). Just add—you
 
 This will prerender your NotFoundPage to `404.html` in your dist folder. Note that there's no need to specify a path.
 
-## Cells, Private Routes, and Dynamic URLs
+## Private Routes
 
-How does Prerendering handle dynamic data? For Cells, Redwood prerenders your Cells' `<Loading/>` component. Similarly, for Private Routes, Redwood prerenders your Private Routes' `whileLoadingAuth` prop:
+For Private Routes, Redwood prerenders your Private Routes' `whileLoadingAuth` prop:
 
-```jsx {1,2}
+```jsx
 <Private >
   // Loading is shown while we're checking to see if the user's logged in
   <Route path="/super-secret-admin-dashboard" page={SuperSecretAdminDashboard} name="ssad" whileLoadingAuth={() => <Loading />} prerender/>
 </Private>
 ```
 
-Right now prerendering won't work for dynamic URLs. We're working on this. If you try to prerender one of them, nothing will break, but nothing happens.
+## Dynamic routes
+
+Let's say you have a route like this
 
-```jsx title="web/src/Routes.js"
+```jsx
 <Route path="/blog-post/{id}" page={BlogPostPage} name="blogPost" prerender />
 ```
 
+To be able to prerender this route you need to let Redwood know what `id`s to use. Why? Because when we are prerendering your pages - at build time - we don't know the full URL i.e. `site.com/blog-post/1` vs `site.com/blog-post/3`. It's up to you to decide whether you want to prerender _all_ of the ids, or if there are too many to do that, if you want to only prerender the most popular or most likely ones.
+
+You do this by creating a `BlogPostPage.routeHooks.js` file next to the page file itself (so next to `BlogPostPage.js` in this case). It should export a function called `routeParameters` that returns an array of objects with the path param name as the key, and the value for the parameter as the object value.
+
+A single Page component can be used for different routes too! Metadata about the current route will be passed as an argument to `routeParameters` so you can return different route parameters depending on what route it is, if you need to. An example will hopefully make all this clearer.
+
+For the example route above, all you need is this:
+
+```js title="BlogPostPage.renderData.js"
+export function routeParameters() {
+  return [{ id: 1 }, { id: 2 }, { id: 3 }]
+}
+```
+
+Or, if you wanted to get fancy
+
+```js title="BlogPostPage.renderData.js"
+export function routeParameters(route) {
+  if (route.name === 'odd') {
+    return [{ id: 1 }, { id: 3 }, { id: 5 }]
+  } else {
+    return [{ id: 2 }, { id: 4 }, { id: 6 }]
+  }
+}
+```
+
+In these routeHooks scripts you have full access to your database using prisma and all your services, should you need it. You use `import { db } from '$api/src/lib/db'` to get access to the `db` object.
```

Comment:
I think it might be work having an example with the db usage. Additionally, we should make it really clear that you need the `$` to import the db. That is something that could be easily missed.

Reply:
I added another example. One that uses the DB, and made a note about the import